src/dbnode/client/connection_pool.go

```diff
@@ -182,34 +182,27 @@ func (p *connPool) connectEvery(interval time.Duration, stutter time.Duration) {
 		var wg sync.WaitGroup
 		for i := 0; i < target-poolLen; i++ {
 			wg.Add(1)
-			newConnFn := p.opts.NewConnectionFn()
 			go func() {
 				defer wg.Done()
 
-				// Create connection
-				channel, client, err := newConnFn(channelName, address, p.opts)
```

Comment:
i don't follow what the original leak was

Reply:
Here there was no leak. I have extracted code into establishConnection and reused it for dedicated session that had a leak before I fixed it here https://github.com/m3db/m3/pull/4026